### PR TITLE
Added notes, restructured project.

### PR DESCRIPTION
**Pull request by herrozerro**
Added Notes, restructured project to make it easier to hook up to different UIs, changed namespaces to match the project name.

Updated to .net framework 4.6.1 to allow for the Core library to be a .net standard 2.0 library.  This will allow for easier segregation of responsibilities.

Created a /src folder and put the solution and then moved the project files into folders.

### CODE REVIEW

**Comment by sander1095** [comment]
Hello! 

I will take a look tonight at this! It's amazing that you chose to contribute to this project! Thanks!

**Comment by sander1095** [comment]
I am currently looking into this, I will paste some of the text I pasted in #27 here too:

> I wanna figure out the best way to merge your PR. Perhaps merging it into develop first for testing purposes and all that would be a better way than merging it to master?

**Comment by herrozerro** [comment]
Do I have to redo this or into develop?

**Comment by sander1095** [comment]
I think merging it into develop would be a better idea. 

I can create a new branch from develop with my current changes and then revert commit the changes I made so far on develop so no broken code will be  in the next release! 

I am quite new to the open source community so I will have to figure out the best way to these things.

I would love to have some CI/CD set up in the future as well, I have experience with Azure DevOps so perhaps we could use that? That can be a seperate issue entirely though.

Again, thank you very much for your contribution!

**Comment by herrozerro** [comment]
No problem, I'm fairly weak in the git department, so I'm not sure the best way to do this.

File: src/DnDCombatTracker.Core/Character.cs [hunk 1]
```diff
@@ -50,6 +52,7 @@ public Character(string name, short initiative, short? maxHP, short? HP, short?
             DeathSaves_Fail = fail;
             DeathSaves_Success = success;
             Conditions = conditions;
+            Note = notes;
```
Comment (sander1095) [inline src/DnDCombatTracker.Core/Character.cs]: Perhaps you could rename the property `Note` to `Notes`, it's a better name. I see your constructor param name is already notes and that fits better.

perhaps you also have to update the UI to reflect this, idk if we are showing 'Note' there right now

Comment (herrozerro) [reply src/DnDCombatTracker.Core/Character.cs]: Sounds good, I can make the changes tonight

Comment (herrozerro) [reply src/DnDCombatTracker.Core/Character.cs]: Nevermind, I resolved it.  Also the GUI is named correctly.

Comment (sander1095) [reply src/DnDCombatTracker.Core/Character.cs]: afbeelding


Seems i found a bug.

I see there are now 2 note fields, one in details, one in the "information" tab.

When I have multiple characters and I select a different one, the information tab updates with the information of the selected character.

The notes do not change, although they should.


Reproduce:

- Create a character
- Add a note
- Save
- Create another character
- Add a different note
- Save
- Click in the list on the other character. all information updates  except the notes

**Comment by sander1095** [comment]
I would recommend to create multiple commits next time so it is easier to reproduce multiple changes. Right now it's difficult to see  what has changed because git says everything is changed because files are now in a new folder. If this would have been a seperate commit I would have been able to see exactly what you changed in the code :)

File: src/DnDCombatTracker.Core/Character.cs (hunk 1 above)
Comment (herrozerro) [reply src/DnDCombatTracker.Core/Character.cs]: I see that, I'll be looking into it.  I think I missed the information panel update

**Comment by herrozerro** [comment]
If it would be better, I could do just that.  Roll Back and reapply my changes, especially the restructure in a separate commit.

File: src/DnDCombatTracker.Core/Character.cs (hunk 1 above)
Comment (herrozerro) [reply src/DnDCombatTracker.Core/Character.cs]: Resolved with the latest commit

**Comment by sander1095** [comment]
That'd be helpful :)!

**Comment by herrozerro** [comment]
I'll redo this and submit a new PR.

**Comment by sander1095** [comment]
I'll wait for the new one and close this one

**Comment by sander1095** [comment]
By supporting a newer framework version, the prerequisites also need to be updated:

Visual Studio -> Forms project -> Properties -> Publish -> Prerequisites -> Enable the new framework version, disable the old one

**Comment by sander1095** [comment]
in your update, I suppose we should also increase the minor version of the project. 

I have been trying to publish the project again from your Pr branch but I have been getting a few errors:

> Error: An error occurred trying to download 'https://raw.githubusercontent.com/sander1095/DnDCombatTracker/master/publish/DnDCombatTracker.application'.


On my master it's called `CombatTracker.application`, a bit odd how it doesn't work anymore.

I don't feel like it has to be fixed in your PR, but if you want to, go ahead. Otherwise we'll fix it on `dev`. 

The auto updater works thanks to this article:

https://refactorsaurusrex.com/post/2015/how-to-host-a-clickonce-installer-on-github/

**Pull request closed without merging**